src/boot/store.js

```diff
@@ -204,6 +227,44 @@ function listMiddleware() {
     // functions of `dispatch` and `state`.  See docs:
     //   https://github.com/reduxjs/redux-thunk
     thunkMiddleware,
+
+    /**
+     * Drop an action from an account that is not the active account.
+     *
+     * Will only be given a plain-object action as input, because of
+     * its placement after `thunkMiddleware`.
+     *
+     * If the action has the special `initiatingIdentity` field, check
+     * it against the identity of the currently active account, and
+     * drop the action if it's not a match.
```

Comment:
Hmm, interesting solution!

I think this is not actually needed for the present use case. IIUC the TYPING_CLIENT_TIMER_EXPIRED action effectively just tells the reducer "hey BTW time T has come and gone, I heard a rumor you might want to update your state at key X at that time, take a look and see." And the reducer won't actually change anything unless it turns out it did indeed want to make that change at time T.

I could imagine cases where this is helpful for something else; say perhaps where we have some code make an API request and then dispatch an action with the resulting data, though that may not be a great example because we hope to cover those more systematically by canceling fetches on logout or account switch.

Perhaps keep this around in your back pocket for if we spot another case where we want to use it?

Reply:
Are you leaning toward it being fine to let the state get updated as normal, without checking whether the same user that received the EVENT_TYPING_START is still logged in?

Maybe that's fine; I guess if there's a rapid user switch and they happen to have a conversation with the exact same people, it wouldn't feel like a glitch if the new user's state was informed by the previous user's state. (A person is either typing in a conversation, or they're not.) And, thinking about privacy of the knowledge that a different account has an active conversation with the same people as the currently active account, it's more reasonable to suppose the two accounts are owned by the same person than it would be if we were building a web app (which people will log into on public computers).

Unfortunately, we can't leave the decision to ignore (or treat differently) actions for no-longer-active accounts up to the individual reducers. `typingReducer` doesn't know anything about the `accounts` state, which we'd need to determine the active account.

Reply:
> And, thinking about privacy of the knowledge that a different account has an active conversation with the same people as the currently active account, it's more reasonable to suppose the two accounts are owned by the same person than it would be if we were building a web app (which people will log into on public computers).

Right -- there's no privacy issue with information moving around the app. A potential privacy issue only arises when sending information to the server (or outside the app in general, but for us that almost always means to the server.)

> Maybe that's fine; I guess if there's a rapid user switch and they happen to have a conversation with the exact same people, it wouldn't feel like a glitch if the new user's state was informed by the previous user's state.

Right but I think even more than that is true here: because the reducer checks the timestamp, the only information the reducer is really relying on from the action is the fact that the current time is at least what the action says it is. If the reducer gets an action which was really intended for the old account but ends up acting on it, then it must also have been the case that it was already late enough that it ought to be getting any moment now an action with the same effect that was intended for the new account.